packages/bpk-component-datatable/src/BpkDataTable.js

```diff
@@ -18,179 +18,222 @@
-import _omit from 'lodash/omit';
 
 import STYLES from './BpkDataTable.module.scss';
-import BpkDataTableColumn from './BpkDataTableColumn';
-import hasChildrenOfType from './hasChildrenOfType';
-import makeSorter, { type Sorter } from './sorter';
 import type { Props } from './common-types';
+import { SORT_DIRECTION_TYPES } from './sort-types';
+import BpkDataTableHeader from './BpkDataTableHeader';
+import { getColumns } from './utils';
 
 const getClassName = cssModules(STYLES);
-const omittedTableProps = ['rowGetter', 'rowCount', 'onHeaderClick'];
 
-type State<Row> = {
-  sorter: Sorter<Row>,
-  rowSelected: ?number,
+const KEYCODES = {
+  ENTER: 13,
+  SPACEBAR: 32,
 };
 
-class BpkDataTable<Row> extends Component<Props<Row>, State<Row>> {
-  static defaultProps = {
-    ...Table.defaultProps,
-    width: null,
-    headerHeight: 60,
-    rowHeight: 60,
-    gridStyle: { direction: undefined }, // This is required for rows to automatically respect rtl
-    defaultColumnSortIndex: 0,
-    sort: null,
-    sortBy: null,
-    sortDirection: null,
-  };
-
-  constructor(props: Props<Row>) {
-    super();
-
-    this.state = {
-      sorter: makeSorter(props),
-      rowSelected: undefined,
-    };
+const BpkDataTable = (props: Props) => {
+  const {
+    children,
+    className,
+    defaultColumnSortIndex,
+    height,
+    rows: data,
+    sort,
+    sortBy,
+    sortDirection,
+    width,
+    ...restOfProps
+  } = props;
+
+  const [rowSelected, updateRowSelected] = useState(undefined);
+
+  const classNames = [getClassName('bpk-data-table')];
+  if (className) {
+    classNames.push(className);
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps: Props<Row>) {
-    if (nextProps.rows !== this.props.rows) {
-      this.state.sorter.propsChange(nextProps);
-    }
-  }
-
-  onRowClicked = ({ index }: { index: number }) => {
-    if (this.state.rowSelected === index) {
-      this.setState({ rowSelected: undefined });
-    } else {
-      this.setState({ rowSelected: index });
-    }
-    if (this.props.onRowClick !== undefined) {
-      this.props.onRowClick(this.state.sorter.getRow(index));
-    }
-  };
-
-  onHeaderClick = ({
-    dataKey: sortBy,
-    event,
-  }: {
-    dataKey: string,
-    event: SyntheticEvent<HTMLDivElement>,
-  }) => {
-    const column = React.Children.toArray(this.props.children).find(
-      (child) => child.props.dataKey === sortBy,
-    );
-
-    if (!column) {
-      return;
-    }
-
-    if (column.props.disableSort === true) {
-      return;
-    }
-
-    // See: https://reactjs.org/docs/legacy-event-pooling.html
-    const eventTarget = event.target;
-    if (eventTarget instanceof Element) {
-      this.setState((prevState) => ({
-        sorter: prevState.sorter.onHeaderClick(sortBy, eventTarget, column),
-      }));
-    }
-  };
-
-  rowClassName = (
+  const getRowClassNames = (
     consumerClassName: ?string | ?(({ index: number }) => string),
-    { index }: { index: number },
+    index: number,
```

Comment:
{ index }: { index: number },

Reply:
ah actually this should be fine as you no longer need to destructure the index because it's passed directly 😬